packages/remix-dev/compiler/utils/tsconfig/write-config-defaults.ts

```diff
@@ -37,15 +38,33 @@ function objectKeys<Type extends object>(value: Type): Array<ObjectKeys<Type>> {
 }
 
 export function writeConfigDefaults(configPath: string) {
-  let configContents = fse.readFileSync(configPath, "utf-8");
-  let config = JSON5.parse(configContents);
-  let configType = path.basename(configPath);
-  if (!config.compilerOptions) {
-    config.compilerOptions = {};
+  // check files exist
+  if (!fse.existsSync(configPath)) return;
+
+  // this will be the *full* tsconfig.json with any extensions deeply merged
+  let fullConfig = loadTsconfig(configPath) as TsConfigJson | undefined;
+  // this will be the user's actual tsconfig file
+  let configContents = fse.readFileSync(configPath, "utf8");
+
+  let config: TsConfigJson | undefined;
+  try {
+    config = JSON5.parse(configContents);
+  } catch (error: unknown) {}
+
+  if (!fullConfig || !config) {
+    // how did we get here?
```

Comment:
Maybe add a `console.warn` saying: "This should never happen, please open an issue with a reproduction." or something.